### PR TITLE
Remove Router + Add Registry + Modify Buffer, IConduit and example

### PR DESCRIPTION
**Pull request by sunbreak1211**

### CODE REVIEW

File: src/AllocatorBuffer.sol [hunk 1]
```diff
@@ -84,15 +69,13 @@ contract AllocatorBuffer is IAllocatorConduit {
         emit Approve(asset, spender, amount);
     }
 
-    function deposit(bytes32, address asset, uint256 amount) external {
+    function deposit(address asset, uint256 amount) external {
```
Comment (oldchili) [inline src/AllocatorBuffer.sol]: Do we still need the `deposit` and `withdraw` functions here?
I'm not sure as it seems the conduits, swapper, and depositor don't use them.
Maybe just for future support if needed?

File: src/AllocatorConduitExample.sol [hunk 2]
```diff
@@ -82,16 +88,18 @@ contract AllocatorConduitExample is IAllocatorConduit {
     // --- functions ---
 
     function deposit(bytes32 ilk, address asset, uint256 amount) external ilkAuth(ilk) {
+        address buffer = registry.buffers(ilk);
+        // Implement the logic to deposit funds from the Buffer into the FundManager
```
Comment (oldchili) [inline src/AllocatorConduitExample.sol]: Can you add the `transferFrom` and `transfer` calls in deposit and withdraw?

File: src/AllocatorBuffer.sol (hunk 1 above)
Comment (sunbreak1211) [reply src/AllocatorBuffer.sol]: Removed.

File: src/AllocatorConduitExample.sol (hunk 2 above)
Comment (sunbreak1211) [reply src/AllocatorConduitExample.sol]: Done.

**Pull request merged**